lib/azuremetadata/azuremetadata.py

```diff
@@ -15,25 +22,58 @@ def get_all(self):
         return result
 
     def get_instance_data(self):
-        # FIXME this is a stub while the CLI UX is being figured out
-
-        if self._api_version == '2019-08-15':
-            with open('fixtures/metadata-v2019-08-15-multiple-network-if.json', 'r') as json_file:
-                return json.load(json_file)
-        elif self._api_version == '2017-04-02':
-            with open('fixtures/metadata-v2017-04-02.json', 'r') as json_file:
-                return json.load(json_file)
-        else:
-            raise Exception("Unknown API version fixture exception")
+        return self._make_request(
+            f"http://169.254.169.254/metadata/instance?api-version={quote(self._api_version)}"
+        )
 
     def get_attested_data(self):
-        # FIXME this is a stub while the CLI UX is being figured out
-
-        if self._api_version == '2019-08-15':
-            with open('fixtures/attested-data-v2019-08-15.json', 'r') as json_file:
-                return json.load(json_file)
-        elif self._api_version == '2018-10-01':
-            with open('fixtures/attested-data-v2018-10-01.json', 'r') as json_file:
-                return json.load(json_file)
-        else:
-            raise Exception("Unknown API version fixture exception (attested data)")
+        return self._make_request(
+            f"http://169.254.169.254/metadata/attested/document?api-version={quote(self._api_version)}"
+        )
+
+    def get_disk_tag(self, device=None):
+        if not device:
+            device = self._find_root_device()
+
+        with open(device, 'rb') as fh:
+            fh.seek(65536)
+            return str(uuid.UUID(bytes_le=fh.read(16)))
+
+    @staticmethod
+    def _find_root_device():
+        hex_device_id = os.stat("/").st_dev
+        root_device_id = str(os.major(hex_device_id)) + ":" + str(os.minor(hex_device_id))
+
+        devices = glob.glob("/sys/block/*")
+        for device_path in devices:
+            device = os.path.basename(device_path)
+            partitions = glob.glob(f"/sys/block/{device}/{device}*")
+
+            for partition in partitions:
+                with open(f"{partition}/dev", "r") as fh:
+                    device_id = fh.read().strip()
+
+                if device_id == root_device_id:
+                    return f"/dev/{device}"
+
+        raise Exception("Can't find root device")
```

Comment:
Not sure if we should have that as [other cases](https://github.com/SUSE-Enceladus/azuremetadata/blob/6d86995cd3bb68756f2d7532bb085e9af270c2e3/azuremetadata#L72)

Reply:
WE should not throw exceptions during data acquisition. It's a best effort implementation, i.e. `azuremetadata --tag --external-IP` should still return the external IP even if we cannot find the root device. The error needs to be noted, but throwing the exception will prevent gathering other requested data.